src/device_batch_trsm.cc

```diff
@@ -84,12 +84,17 @@ void trsm(
             std::swap( m_, n_ );
         }
 
-        size_t batch_limit = queue.get_batch_limit();
-        scalar_t** dAarray = (scalar_t**) queue.get_dev_ptr_array( );
-        scalar_t** dBarray = dAarray + batch_limit;
+        // trsm needs only 2 ptr arrays (A and B). Allocate usual
+        // 3*max_chunk used by gemm, but then split in 2 instead of 3.
+        size_t max_chunk = MaxBatchChunk;
+        queue.work_resize<void*>( 3*max_chunk );
+        max_chunk = 3*max_chunk / 2;
```

Comment:
Why give 3/2 of max_chunk to a matrix when, from gemm point of view, one max_chunk is used?